### PR TITLE
Specify more details of the extend algorithm

### PR DESCRIPTION
**Pull request by nex3**
This lays down some specs which can then be updated for #2829.

### CODE REVIEW

File: spec/at-rules/extend.md [hunk 1]
```diff
@@ -216,6 +221,118 @@ that includes CSS for *all* modules transitively used or forwarded by
...
+
+* Let `results` be an empty selector list.
+
+* For each complex selector `complex` in `extendee`:
+
+  * Let `options` be an empty complex selector.
+
+  * For each compound selector `compound` or combinator in `complex`:
+
+    * If it's a combinator, add it to each selector in `complex-options`.
```
Comment (jathak) [inline spec/at-rules/extend.md]: Should `complex-options` here be `options`?

File: spec/at-rules/extend.md [hunk 2]
```diff
@@ -216,6 +221,118 @@ that includes CSS for *all* modules transitively used or forwarded by
...
+
+    * Let `new-complex-options` be an empty list.
```
Comment (jathak) [inline spec/at-rules/extend.md]: This is never added to or referenced from what I can tell

**Review by jathak — COMMENTED** [review]
The spec for extending simple selectors and the syntax looks good, but I couldn't quite parse what was going on with the complex selector case.

**Comment by nex3** [comment]
> The spec for extending simple selectors and the syntax looks good, but I couldn't quite parse what was going on with the complex selector case.

I added an example, does that help?

**Review by jathak — APPROVED** [review]
Yes, that helps. LGTM.

**Pull request merged**